clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -235,6 +235,33 @@ func getSelectorFromUnstrutured(obj *unstructured.Unstructured) (labels.Selector
 	}
 }
 
+func GetUpdatedPodPredicateFromRuntimeObject(obj runtime.Object) (func(_ *corev1.Pod) bool, error) {
+	switch typed := obj.(type) {
+	case *unstructured.Unstructured:
+		return getUpdatedPodPredicateFromUnstructured(typed)
+	default:
+		return nil, fmt.Errorf("unsupported kind when getting updated pod predicate: %v", obj)
```

Comment:
NIT: Can you use goerrors? (example [here](https://github.com/kubernetes/perf-tests/blob/75dfad9672dfabe11b292c246f8f99c79db84655/clusterloader2/pkg/config/template_provider.go#L225))

The syntax is the same as fmt.Errorf, but it gives you stack-traces. Here and in other places if it's not too much work

Reply:
Done